### PR TITLE
Add tests for FileSettingsService.handleSnapshotRestore

### PR DESCRIPTION
**Pull request by thecoop**
Streamline FileSettingsService tests a bit - don't create duplicate objects during tests

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/reservedstate/service/FileSettingsServiceTests.java [hunk 1]
```diff
@@ -229,36 +220,77 @@ public void testStopWorksInMiddleOfProcessing() throws Exception {
...
+                    .build()
+            )
+            .build();
+
+        Metadata.Builder metadata = Metadata.builder();
+        fileSettingsService.handleSnapshotRestore(state, metadata);
+
+        assertThat(
+            metadata.build().reservedStateMetadata(),
+            hasEntry(FileSettingsService.NAMESPACE, new ReservedStateMetadata(FileSettingsService.NAMESPACE, 0L, Map.of(), null))
```
Comment (mosche) [inline server/src/test/java/org/elasticsearch/reservedstate/service/FileSettingsServiceTests.java:290]: nit, for readability?
```suggestion
            hasEntry(FileSettingsService.NAMESPACE, new ReservedStateMetadata(FileSettingsService.NAMESPACE, ReservedStateMetadata.RESTORED_VERSION, Map.of(), null))
```

**Review by mosche — APPROVED** [review]
lgtm

**Comment by thecoop** [comment]
@elasticmachine rerun elasticsearch-ci/part-2

**Comment by thecoop** [comment]
@elasticmachine update branch

**Pull request merged**